src/libs/actions/connections/QuickBooksOnline.ts

```diff
@@ -0,0 +1,11 @@
+import {getCommandURL} from '@libs/ApiUtils';
+
+function getQuickBooksOnlineSetupLink(policyID: string) {
```

Comment:
@mountiny suggested that we might have to clear `categories` and `tags` optimistically (when accounting software is connected, we want to use imported values for these data. so we have to clear them. there is already a code to clear these values in the backend, but we might want to do so optimistically on the front end)

Reply:
thinking about that again, I think we shouldn't do it because the flow starts by opening the page `https://www.expensify.com.dev/api/ConnectPolicyToQuickbooksOnline` in a new tab (or within a a WebView). That page doesn't really respond with anything and ends up redirecting to the QBO authentication page.

If you clear optimistically and the user just closes the tab that opened, you won't have anyway of knowing about the failure so won't be able to revert the optimistic change.

Also, the authentication flow can't be done without internet anyway, so it has little value to do optimistic stuff.

Once the connection is successful and the first import is done, the `SavePolicy` Auth command will generate the onyx updates that will replace the categories and tags by the new ones.

Reply:
yes there is great possibility that the user won't finish the authentication flow or the authentication flow fails.

I agree. Let's not optimistically clear the categories and tags. It won't work in this case 👍 .

Reply:
Yeah I agree with you after thinking about this flow more! Thanks for looking into this 🙇